pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -217,6 +217,10 @@ func (r *ReconcileMemberList) Reconcile(request reconcile.Request) (reconcile.Re
 			}
 			break
 		}
+
+		// tell Kiali that MT mode is disabled. This allows Kiali to see the full cluster.
+		r.reconcileKiali(instance.Namespace, nil, reqLogger)
```

Comment:
I think this is an incorrect assumption.  Just because the smmr is being deleted does not mean the control plane is becoming cluster wide.  I think two things are true here:

1. the user deleted the smmr
2. the user deleted the smcp (we add an owner ref to the smcp, so cleanup of smmr occurs when the control plane is deleted).

Furthermore, there's a fringe case where the user may convert a multitenant install into a cluster wide install (although, this is probably something we should prevent in the validator).

Reply:
So what is the semantics here wrt how Kiali should behave/be configured? Do we just tell Kiali it cannot see any namespaces (other than the control plane namespace)? That would be a simple fix.

>  there's a fringe case where the user may convert a multitenant install into a cluster wide install

OK, that is what this code (line 222) is thinking this means. So you are saying this is NOT converting a multi-tenant install to a cluster-wide install and that we should prevent that from ever happening. Then that means I'll need an answer to the above question - what is the semantics as it is related to Kiali? I think  you are saying if the SMMR is deleted (thus no mesh namespaces are provided) then Kiali should not have any access to any namespace outside the control plane namespace.

Reply:
> So what is the semantics here wrt how Kiali should behave/be configured? Do we just tell Kiali it cannot see any namespaces (other than the control plane namespace)? That would be a simple fix.

Yes.

> OK, that is what this code (line 222) is thinking this means. So you are saying this is NOT converting a multi-tenant install to a cluster-wide install and that we should prevent that from ever happening. Then that means I'll need an answer to the above question - what is the semantics as it is related to Kiali? I think you are saying if the SMMR is deleted (thus no mesh namespaces are provided) then Kiali should not have any access to any namespace outside the control plane namespace.

Adding some color here: it is possible that the user can convert (although we should prevent in the future) a multitenant install into a cluster wide install.  In that case, the smmr reconciler will remove any configured members from the mesh (this is really just cleaning up the work done when it was multitenant, e.g. removing role bindings, etc.).  For kiali, you can use the `isMultitenant` flag and set the namespaces to `**` if it's false, or you could do nothing (assumption being the kiali resource was already updated to be cluster wide by the smcp reconciliation process).

Reply:
Rather than passing in an nil value, code now passes in the control plan namespace. This makes it clear what the semantics are here - at this point in the reconciliation process, we are telling kiali to only have access to the control plane namespace and no others. See: https://github.com/Maistra/istio-operator/pull/87/files#diff-3029a0e7a4af7d79dcfe22782cf7dbbaR222